### PR TITLE
feat: Add conformance test c001

### PR DESCRIPTION
**Pull request by Rqnsom**
- Added a handshake test where the synthetic node initiates a connection towards the node. The test expects the handshake procedure to succeed.
- Added basic Synthetic node impl
- WebSocket codec impl
- Synthetic node tracing impl
- Implemented pea2pea handshake trait procedure for the initiator side.

### CODE REVIEW

**Review by Rqnsom — COMMENTED** [review]
This commit is based on @ljedrz's handshake implementation [here](https://github.com/ljedrz/pea2pea/blob/algo/examples/algo.rs).
He takes all the credit. 🙂

File: src/protocol/handshake.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+struct SecWebSocket {
+    key: String,
+    accept: String,
+}
+
+impl SecWebSocket {
+    /// Generate key-accept pair for a WebSocket handshake.
+    fn generate() -> Self {
+        // TODO(Rqnsom): Randomize 16 bytes.
+        let key: Vec<u8> = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
```
Comment (ljedrz) [inline src/protocol/handshake.rs:33]: we should probably use a more randomized key to have greater coverage in the tests

Comment (ljedrz) [reply src/protocol/handshake.rs:33]: ah, it's a TODO :+1:

**Review by ljedrz — APPROVED** [review]
LGTM :ok_hand:

File: src/protocol/handshake.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        match node_conn_side {
+            ConnectionSide::Initiator => {
+                let mut framed = Framed::new(stream, BytesCodec::default());
+                let sec_ws = SecWebSocket::generate();
+
+                let mut req = Vec::new();
+                req.extend_from_slice(b"GET /v1/private-v1/gossip HTTP/1.1\r\n");
+                req.extend_from_slice(format!("Host: {}\r\n", conn_addr).as_bytes());
+                req.extend_from_slice(
+                    b"User-Agent: algod/3.9 (stable; commit=921e8f6f+; 0) linux(amd64)\r\n",
```
Comment (ljedrz) [inline src/protocol/handshake.rs:58]: note for later: this should probably be delegated to some const, so we can update the client version more easily

File: src/protocol/handshake.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+                req.extend_from_slice(b"Connection: Upgrade\r\n");
+                req.extend_from_slice(format!("Sec-WebSocket-Key: {}\r\n", sec_ws.key).as_bytes());
+                req.extend_from_slice(b"Sec-WebSocket-Version: 13\r\n");
+                req.extend_from_slice(b"Upgrade: websocket\r\n");
+                req.extend_from_slice(b"X-Algorand-Accept-Version: 2.1\r\n");
+                req.extend_from_slice(b"X-Algorand-Instancename: synth_node\r\n");
+                req.extend_from_slice(b"X-Algorand-Location: \r\n");
+                req.extend_from_slice(b"X-Algorand-Noderandom: cGVhMnBlYQ==\r\n");
+                // req.extend_from_slice(b"X-Algorand-Telid: d12c01a5-4ca4-4be3-a394-68c8913f3883\r\n"); // TODO: Investigate more
+                req.extend_from_slice(b"X-Algorand-Version: 2.1\r\n");
```
Comment (ljedrz) [inline src/protocol/handshake.rs:69]: ditto

File: src/protocol/handshake.rs (hunk 2 above)
Comment (Rqnsom) [reply src/protocol/handshake.rs:58]: Okay, I'll do it in the next PR where I'll again touch this file anyway

**Pull request merged**